mat64/dense_test.go

```diff
@@ -1633,6 +1634,56 @@ func TestInverse(t *testing.T) {
 			tol: 1e-14,
 		},
 		{
+			a: NewDense(3, 3, []float64{
+				8, 1, 6,
+				3, 5, 7,
+				4, 9, 2,
+			}).T(),
+			want: NewDense(3, 3, []float64{
+				0.147222222222222, -0.144444444444444, 0.063888888888889,
+				-0.061111111111111, 0.022222222222222, 0.105555555555556,
+				-0.019444444444444, 0.188888888888889, -0.102777777777778,
+			}).T(),
+			tol: 1e-14,
+		},
+
+		// This case does not fail, but we do not guarantee that. The success
+		// is because the receiver and the input are aligned in the call to
+		// inverse. If there was a misalignment, the result would likely be
+		// incorrect and no shadowing panic would occur.
+		{
+			a: asBasicMatrix(NewDense(3, 3, []float64{
+				8, 1, 6,
+				3, 5, 7,
+				4, 9, 2,
+			})),
+			want: NewDense(3, 3, []float64{
+				0.147222222222222, -0.144444444444444, 0.063888888888889,
+				-0.061111111111111, 0.022222222222222, 0.105555555555556,
+				-0.019444444444444, 0.188888888888889, -0.102777777777778,
+			}),
+			tol: 1e-14,
+		},
+
+		// The following test will fail since it does not obey the shadowing
+		// rules. The basicMatrix type allows shadowing of the input data
+		// without providing the Raw mathod required for detection of shadowing.
```

Comment:
This is a bit of a contrived example, which is fine, but I think the comment could be adjusted to make that more clear. 

"The following case fails as it disobeys the shadowing rules. Specifically, the test below extracts the underlying *Dense, and uses it as a recevier with the basicMatrix as input. The basicMatrix type allows shadowing of the input data without providing the Raw mathod required for detection of shadowing. We specifically state we cannot check this case."